units/commit_queue.go

```diff
@@ -306,6 +306,7 @@ func (j *commitQueueJob) processGitHubPRItem(ctx context.Context, cq *commitqueu
 		j.logError(err, "can't insert patch", nextItem)
 		j.dequeue(cq, nextItem)
 		j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't make patch", ""))
+		return
 	}
 	nextItem.Version = patchDoc.Id.Hex()
 	if err = cq.UpdateVersion(nextItem); err != nil {
```

Comment:
should we return here too?

Reply:
I don't think so, and I think that wasn't your intention for a reason (we also don't dequeue or add an error here), because this isn't strictly necessary, it's more for organizational purposes, but if you disagree I don't feel strongly.

Reply:
I feel like an item in the queue without a version will throw a wrench into assumptions we make about the item.
For example, we use it to determine if a version exists for the item when we [preventMergeForItem](https://github.com/evergreen-ci/evergreen/blob/89f2312f6dcbb078f7007c58d0da8806304cc82e/model/commitqueue/commit_queue.go#L214). (This is an argument for dequeueing and returning here)

Reply:
Okay that's reasonable.